api/dp-openapi.yaml

```diff
@@ -678,6 +678,315 @@ paths:
         default:
           $ref: '#/components/responses/UnexpectedError'
 
+  /projects/{projectid}/checkout:
+    put:
+      tags:
+      - project
+      description: Check out some pages for proofreading
+      parameters:
+      - $ref: '#/components/parameters/projectid'
+      - $ref: '#/components/parameters/projectstate'
+      - name: npage
+        in: query
+        description: number of pages to checkout
```

Comment:
There were two motivations for checking out multiple pages at a time.

- Ordinary proofreaders often like to work on consecutive pages and it is possible to checkout several pages quickly in the present system by refreshing the proofreading page. This is probably not something we want to do so to pre-empt complaints about reducing functionality we can have a way to checkout several pages. We could do this by checking out one at a time and let the client application call the function several times. The server would still have to impose a limit on how many can be checked out at once and it would still be best to have different functions for checking out and getting the page data to avoid excessive traffic.
- Mentors need to check out all the pages for one proofreader. Finding the 'oldest proofreader' and checking out a block seemed like a simple way to do it. If you only checkout one page then the 'oldest' proofreader can then become another person so you would have to have a mechanism for selecting a particular proofreader.

> Checking out one page at a time has the benefit that we can mitigate a denial of service (someone checking out all of the pages in a round) because we limit the number of API calls a user can do per unit of time.

This will not be a problem because it is not possible to checkout another block of pages if you already have some checked out.

Reply:
> What's the use-case for checking out multiple pages at a time and why can't we fulfill it via multiple single calls? I see that the implementation has `MAX_REQUESTED_PAGES` to restrict how many pages a single call can check out, but I'm not clear on why we need more than one to begin with.

You are right. I will revise this to make it work with single  page checkout.

The mentors use-case is a bit different but I'll consider it further.